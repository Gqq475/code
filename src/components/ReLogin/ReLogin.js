import React from 'react'
import classes from './ReLogin.scss'
import { connect } from 'react-redux'
import { assign } from 'lodash'
import { Button, Checkbox, Dialog } from '@blueprintjs/core'
import { authenticate } from 'actions'
type Props = {
  authenticate: Function,
  authFailureMessage: String
};
export class ReLogin extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.onBtnClick = this.onBtnClick.bind(this)
    this.togglePasswordKeeping = this.togglePasswordKeeping.bind(this)
    this.handleUsernameChange = this.handleUsernameChange.bind(this)
    this.handlePasswordChange = this.handlePasswordChange.bind(this)
    this.state = {
      rememberMe: true,
      username: '',
      password: ''
    }
  }

  onBtnClick () {
    this.props.authenticate(this.state.username, this.state.password, this.state.rememberMe)
  }

  togglePasswordKeeping (e) {
    this.setState(assign({}, this.state, { rememberMe: !this.state.rememberMe }))
  }

  handleUsernameChange (e) {
    this.setState(assign({}, this.state, { username: e.target.value }))
  }

  handlePasswordChange (e) {
    this.setState(assign({}, this.state, { password: e.target.value }))
  }

  render () {
    return (
      <div className={classes['ReLogin-container']}>
        <Dialog className={classes.dialog} isOpen>
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>登录</h6>
          </div>
          <div className={classes.dialogBody}>
            <div className={classes.form}>
              <form className='login-form'>
                <center className={classes.message}>BIF-Quality</center>
                <br />
                <label className={classes.submessage}>请登录</label>
                <span style={{
                  color: 'red'
                }}>{this.props.authFailureMessage}</span>
                <br />
                <br />
                <input type='text' placeholder='账号'
                  value={this.state.username}
                  onChange={this.handleUsernameChange}
                  />
                <br />
                <input type='password' placeholder='密码'
                  value={this.state.password}
                  onChange={this.handlePasswordChange} />
                <br />
                <Checkbox className={classes.checkbox} checked={this.state.rememberMe}
                  label='记住我' onChange={this.togglePasswordKeeping} />
                <Button className='pt-intent-primary pt-large' onClick={this.onBtnClick}>
                  登录
                </Button>
                <div className={classes.create}>注册账号</div>
              </form>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    token: state.auth.token,
    authFailureMessage: state.auth.failureMessage
  }
}

const mapDispatchToProps = (dispatch) => ({
  authenticate: (username, password, rememberMe) => {
    if (rememberMe) {
      localStorage.setItem('rememberMe', username)
    } else {
      localStorage.removeItem('rememberMe')
    }
    dispatch(authenticate(username, password, 'ReLogin'))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(ReLogin)
