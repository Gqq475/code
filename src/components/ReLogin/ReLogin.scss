@import url(https://fonts.googleapis.com/css?family=Roboto:300);
.ReLogin-container {
  
}
  .headerTitle {
    margin: 10px 0px 10px 15px;
    font-size: 16px;
  }
  .logo {
    position: fixed;
    top: 10%;
    left: 10%;
    height: 40px;
  }

  select {
    margin-right: 140px;
    color: #FFFFFF;
    background: #0087DC;
  }

  .form {
    background: #FFFFFF;
    width: 500px;
    margin-bottom: 10px;
    padding: 45px;
  }

  .form input {
    font-family: "Roboto", sans-serif;
    outline: 0;
    width: 60%;
    border: 0;
    line-height: 25px;
    margin: 0 25% 15px;
    box-sizing: border-box;
    font-size: 14px;
    border-bottom: 1px solid #0087DC;
  }

  .form .checkbox {
    margin: 0 25% 15px;
  }

  .form button {
    font-family: "Roboto", sans-serif;
    outline: 0;
    width: 20%;
    border: 0;
    margin: 0 40%;
    color: #FFFFFF;
    background: #0087DC;
    font-size: 14px;
    -webkit-transition: all 0.3 ease;
    transition: all 0.3 ease;
    cursor: pointer;
  }

  .form button:hover,.form button:active,.form button:focus {
    background: #0073BB;
  }

  .form .message {
    font-family: "Roboto", sans-serif;
    font-size: 28px;
    font-weight: bold;
    color: #0087DC;
  }
  .form .submessage {
    font-family: "Roboto", sans-serif;
    font-size: 18px;
    margin: 0 25% 15px;
    color: #0087DC;
  }
  .create {
    display: block;
    width: 60px;
    margin: 0 auto;
    margin-top: 20px;
    color: #0087DC;
    cursor: pointer;
  }
  .footerText {
    color: #FFFFFF;
  }
  
