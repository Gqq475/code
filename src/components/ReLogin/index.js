import ReLogin from './ReLogin'
export default ReLogin
