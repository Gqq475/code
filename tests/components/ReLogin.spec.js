import React from 'react'
import { ReLogin } from 'components/ReLogin/ReLogin'

describe('(Component) ReLogin', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <ReLogin />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
