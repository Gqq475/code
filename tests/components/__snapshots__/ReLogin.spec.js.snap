// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ReLogin render normally 1`] = `
<div
  className="ReLogin-container"
>
  <Dialog
    className="dialog"
    isOpen={true}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        登录
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <div
        className="form"
      >
        <form
          className="login-form"
        >
          <center
            className="message"
          >
            BIF-Quality
          </center>
          <br />
          <label
            className="submessage"
          >
            请登录
          </label>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          />
          <br />
          <br />
          <input
            onChange={[Function]}
            placeholder="账号"
            type="text"
            value=""
          />
          <br />
          <input
            onChange={[Function]}
            placeholder="密码"
            type="password"
            value=""
          />
          <br />
          <Blueprint.Checkbox
            checked={true}
            className="checkbox"
            label="记住我"
            onChange={[Function]}
          />
          <Blueprint.Button
            className="pt-intent-primary pt-large"
            onClick={[Function]}
          >
            登录
          </Blueprint.Button>
          <div
            className="create"
          >
            注册账号
          </div>
        </form>
      </div>
    </div>
  </Dialog>
</div>
`;
